packages/flutter_chat/lib/src/flutter_chat_navigator_userstories.dart

```diff
@@ -0,0 +1,140 @@
+// SPDX-FileCopyrightText: 2023 Iconica
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+import "package:chat_repository_interface/chat_repository_interface.dart";
+import "package:flutter/material.dart";
+import "package:flutter_chat/src/config/chat_options.dart";
+import "package:flutter_chat/src/routes.dart";
+import "package:flutter_chat/src/services/pop_handler.dart";
+import "package:flutter_chat/src/util/scope.dart";
+
+/// Default Chat Userstory that starts at the chat list screen.
+class FlutterChatNavigatorUserstory extends BaseChatNavigatorUserstory {
+  /// Constructs a [FlutterChatNavigatorUserstory].
+  const FlutterChatNavigatorUserstory({
+    required super.userId,
+    required super.options,
+    super.onExit,
+    super.key,
+  });
+
+  @override
+  State<BaseChatNavigatorUserstory> createState() =>
+      _FlutterChatNavigatorUserstoryState();
+}
+
+class _FlutterChatNavigatorUserstoryState
+    extends _BaseChatNavigatorUserstoryState<FlutterChatNavigatorUserstory> {
+  @override
+  MaterialPageRoute initialRoute() => chatOverviewRoute(
+        userId: widget.userId,
+        chatService: _service,
+        chatOptions: widget.options,
+        onExit: widget.onExit,
+      );
+}
+
+/// Chat Userstory that starts directly in a chat detail screen.
+class FlutterChatDetailNavigatorUserstory extends BaseChatNavigatorUserstory {
+  /// Constructs a [FlutterChatDetailNavigatorUserstory].
+  const FlutterChatDetailNavigatorUserstory({
+    required super.userId,
+    required super.options,
+    required this.chat,
+    super.onExit,
+    super.key,
+  });
+
+  /// The chat to start in.
+  final ChatModel chat;
```

Comment:
This should always be an identifier, and it should not necessarily mean a chat id. In beep's case we know it is a chat, because the chat already exists. Imagine on a site that requires a customer service chat, you would want to have this be a ticket id or maybe you simply want to start a chat with another user, in that case it is a user-id.

We should resolve which chat this is in user-story

Reply:
Because this is a rework of how the chatDetailScreen works I will do this in a seperate commit at the end (instead of making the existing commit even bigger